File: 2048-build-array-from-permutation/build-array-from-permutation.cpp
class Solution {
public:
    vector<int> buildArray(vector<int>& nums) {
    vector<int>res;
    int n=nums.size();
    int i=0;
    while(i<nums.size())
    {
        res.push_back(nums[nums[i]]);
        i++;
    }
    return res;
    }
};

File: 2048-build-array-from-permutation/build-array-from-permutation.java
class Solution {
    public int[] buildArray(int[] nums) {
        int n=nums.length;
        int[] res=new int[n];
        int i=0;
        while(i<n)
        {
            res[i]=nums[nums[i]];
            i++;
        }
        return res;
    }
}